### PR TITLE
fix: ProseCode class attribute warning

### PR DESCRIPTION
**Pull request by bobby169**
https://github.com/nuxt/content/issues/2111

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by narasimhajupally** [comment]
we can just wrap the slot in <div> making it root node, doing so the class prop will be automatically inherited.

File: src/runtime/components/Prose/ProseCode.vue [hunk 1]
```diff
@@ -22,13 +22,19 @@ export default defineComponent({
     meta: {
       type: String,
       default: null
+    },
+    class: {
+      type: String,
+      default: "",
```
Comment (nobkd) [inline src/runtime/components/Prose/ProseCode.vue:28]:
```suggestion
```

File: src/runtime/components/Prose/ProseCode.vue [hunk 2]
```diff
@@ -22,13 +22,19 @@ export default defineComponent({
...
     }
   }
 })
 </script>
 
 <template>
-  <slot />
+  <div :class="class">
```
Comment (nobkd) [inline src/runtime/components/Prose/ProseCode.vue:35]:
```suggestion
  <div>
```

**Comment by bobby169** [comment]
> we can just wrap the slot in `<div>` making it root node, doing so the class prop will be automatically inherited.

I try to remove `:class="class"` from root div element, the  class will not be automatically inherited.

image


Add `:class="class"` the result is:

image

With the `class` prop We can define more styling code.

**Comment by farnabaz** [comment]
Thanks for the PR.
Instead of wrapping code, a better solution would be moving class attribute from component itself into inner `code` tag (` pre > code`)

from:
https://github.com/nuxt/content/blob/d7bccdc5ba4c5348d30ed49ba7ea6cf43cc19462/src/runtime/markdown-parser/handler/code.ts#L27-L29

to:
```ts
[h(node, 'pre', {}, [h(node, 'code', { __ignoreMap: '', className: [`language-${language}`] }, [u('text', code)])])]
```

**Comment by wJoenn** [comment]
@farnabaz I just tried your solution locally and it doesn't solve Vue's warning.
Bobby's fix by wrapping the slot inside a div does on the other hand.

Wouldn't it be just fine to merge this as is ?

It could even serve a second purpose by adding a span with the filename to be able to display the filename in the corner of the code block
```
<template>
  <div class="codeBlock">
    <span class="filename">{{ filename }}</span>
    <slot />
  </div>
</template>

<style>
.codeBlock {
  position: relative;
}

.codeBlock .filename {
  font-size: 0.8em;
  position: absolute;
  right: 20px;
  top: 10px;
}

.codeBlock pre code .line {
  display: block;
  min-height: 1rem;
}
</style>
```

image

**Comment by farnabaz** [comment]
Having filename indeed is a good thing to have, but also some user may not want to have a wrapper tag. 
As I checked a couple of websites, they add the class into `pre` tag. Check https://shiki.matsu.io/

Also, I've checked it locally, adding class to `pre` tag resolves the warning 🙂

**Comment by 1714080902120** [comment]
I got the same problem, i updated my `@nuxt/content` version to the latest just now, but it still not work.
image

**Comment by farnabaz** [comment]
I'm closing this in favor of #2187.
This issue is fixed in nuxt-mdc package

**Pull request closed without merging**